posix/posix.c

```diff
@@ -1626,6 +1630,25 @@ int posix_connect(int socket, const struct sockaddr *address, socklen_t address_
 }
 
 
+int posix_gethostname(char *name, size_t namelen)
+{
+	TRACE("gethostname(%zu)", namelen);
+
+	if (namelen < 0)
+		return -EINVAL;
+
+	if (namelen > hostname_len) {
+		hal_memcpy(name, hostname, hostname_len);
+		name[hostname_len] = '\0';
+	}
+	else {
+		hal_memcpy(name, hostname, namelen);
+	}
```

Comment:
My suggestion (hostname_len not needed):

```suggestion
	hal_strncpy(name, hostname, namelen);
	name[namelen] = '\0';
```

Reply:
Your version is shorter but it always adds 0 at the end (POSIX is not clear on this, needs check for namelen == 0). It seems that ia32 hal_strncpy() doesn't handle 0 case properly, I'm not sure about other targets. Also sethostname doesn't guarantee that hostname is NULL terminated so hal_strncpy may copy all HOST_NAME_MAX bytes. I would prefer to keep this version.

Reply:
You can either keep one char extra for NULL terminator or 4 bytes (size_t) for string length. Code turns out better for 1st option imho, but it's a shame, that hal_strncpy() seems to be broken (it would be better to fix it rather than avoid using it).